scripts/experimental/install_dev_osgeo.sh

```diff
@@ -6,7 +6,7 @@ set -e
 ## 'latest' means installing the latest release version.
 
 ## build ARGs
-NCPUS=${NCPUS:-"-1"}
+NCPUS=${NCPUS:-"1"}
```

Comment:
I thought this was for `install2.r`.
Is there any reason to use this with cmake?
Is there a problem with the default value of cmake?

Reply:
Right, I think this was initially introduced for `install2.r`, but compiling OSGEO C libraries from source is perhaps even more time consuming, and definitely benefits from parallelization.  `cmake` errors with the default value of `-1`.  

I suppose we could use separate env vars to define the number of cores available to `install2.r` and the number of cores available to `cmake`, but I suspect that would merely confuse users and developers more?  

Maybe it would be better to auto-detect available cores? Could be more fragile though.

Reply:
I am not sure if this is used by the users. Is there any use case other than `-1` to install2.r?

I disagree with the changing default value to `1` as it takes longer to run on CI.
How about having the command get the number of available cores if `-1` is set?

Reply:
Yes, that's a very good point.  

I guess

```
nproc --all
```

is the best way to do that?

Reply:
> I guess
> 
> ```
> nproc --all
> ```
> 
> is the best way to do that?

I am not familiar with the commands to do this but it seems good!

Reply:
`nproc` part of coreutils in linux, so should be pretty widely available.  I've pushed a change for this now (below); though no doubt the linter won't like my bash :-)